src/frontend/src/flows/recovery/confirmSeedPhrase.ts

```diff
@@ -12,7 +12,7 @@ import copyJson from "./confirmSeedPhrase.json";
 // A list of words, where "check" indicates if the user needs to double check (re-input) a word
 type Word = { word: string } & (
   | { check: false }
-  | { check: true; elem: Ref<HTMLInputElement> }
+  | { check: true; elem: Ref<HTMLInputElement>; shouldFocus?: boolean }
```

Comment:
What happens if I pass `shouldFocus: true` to more than one word in `confirmSeedPhraseTemplate`?

Reply:
only the first one will focus: https://cdpn.io/pen/debug/eYLLMLQ/bf640e6faa44d15f891c9e1d55b7f9d2 (so theoretically we could all set them to autofocus :D but I would find this strange plus it would go against the spec) MDN sais:

> No more than one element in the document or dialog may have the autofocus attribute. If applied to multiple elements the first one will receive focus.

Reply:
heh sorry that was a rhetorical question. What I meant is: why does the caller of `confirmSeedPhraseTemplate` need to care about this? Also why is `shouldFocus` optional? see what we do with `elem`, which (like `shouldFocus`) is data internal to the function, but added to `Word` for convenience.

I suspect `shouldFocus` was made optional just because, when calling `confirmSeedPhraseTemplate`, it is not set; but really it shouldn't even be "settable" at all.

Reply:
I didn't try running this, but I think you can easily mark `shouldFocus` an not optional, and avoid `"elem" in ...` bit:

```diff
diff --git a/src/frontend/src/flows/recovery/confirmSeedPhrase.ts b/src/frontend/src/flows/recovery/confirmSeedPhrase.ts
index 7339f9ee..de4288ec 100644
--- a/src/frontend/src/flows/recovery/confirmSeedPhrase.ts
+++ b/src/frontend/src/flows/recovery/confirmSeedPhrase.ts
@@ -12,7 +12,7 @@ import copyJson from "./confirmSeedPhrase.json";
 // A list of words, where "check" indicates if the user needs to double check (re-input) a word
 type Word = { word: string } & (
   | { check: false }
-  | { check: true; elem: Ref<HTMLInputElement>; shouldFocus?: boolean }
+  | { check: true; elem: Ref<HTMLInputElement>; shouldFocus: boolean }
 );
 
 // A list of indices nicely spread over the 25 words (anchor + 24 BIP39)
@@ -31,7 +31,7 @@ const confirmSeedPhraseTemplate = ({
   back,
   i18n,
 }: {
-  words: Omit<Word, "elem">[];
+  words: Omit<Word, "elem" | "shouldFocus">[];
   confirm: () => void;
   back: () => void;
   i18n: I18n;
@@ -47,16 +47,18 @@ const confirmSeedPhraseTemplate = ({
         word: word.word,
         check: word.check,
         elem,
+        shouldFocus: false,
       };
     } else {
       return { word: word.word, check: word.check };
     }
   });
 
-  // find the first editable word and focus it
-  const firstEditable = words.find((w) => w.check);
-  if (firstEditable && "elem" in firstEditable) {
-    firstEditable.shouldFocus = true;
+  for(const word of words) {
+      if(word.check) {
+          word.shouldFocus = true;
+          break;
+      }
   }
 
   // if all "check" words have been re-input correctly
```

Reply:
good point, thats much nicer.